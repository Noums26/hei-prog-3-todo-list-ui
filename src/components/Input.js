import React, { useState } from "react";

const Input = ({ sendTodos }) => {
  const [todo, setTodo] = useState("");

  const submitTodo = (e) => {
    e.preventDefault();
    todo && sendTodos({ message: todo, state: false });
    setTodo("");
  };

  return (
    <form onSubmit={submitTodo}>
      <input
        type="text"
        onChange={(e) => setTodo(e.target.value)}
        value={todo}
        className="todo-input"
        placeholder="Add todo"
      />
    </form>
  );
};

export default Input;
