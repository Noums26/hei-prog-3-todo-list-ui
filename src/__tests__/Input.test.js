import renderer from "react-test-renderer";
import Input from "../components/Input";

describe("Input test", () => {
  test("should test input snapshot", () => {
    const input = renderer.create(<Input sendTodos={jest.fn()} />).toJSON();

    expect(input).toMatchSnapshot();
  });
});
