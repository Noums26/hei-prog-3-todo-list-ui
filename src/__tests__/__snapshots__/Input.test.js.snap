// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input test should test input snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    className="todo-input"
    onChange={[Function]}
    placeholder="Add todo"
    type="text"
    value=""
  />
</form>
`;
